src/services/hostService.ts

```diff
@@ -78,4 +80,18 @@ export class HostService extends Disposable implements IHostService {
         }
         return true;
     }
+
+    // Force Linux as host platform (https://github.com/gitpod-io/gitpod/issues/16058)
+    async updateSSHRemotePlatform() {
+        try {
+            const hostname = '*.' + (new URL(this.gitpodHost)).hostname;
+            const existingSSHHostPlatforms = vscode.workspace.getConfiguration('remote.SSH').get<{ [host: string]: string }>('remotePlatform', {});
+            const targetPlatform = 'linux';
+            if (!existingSSHHostPlatforms[hostname] || existingSSHHostPlatforms[hostname] !== targetPlatform) {
+                await vscode.workspace.getConfiguration('remote.SSH').update('remotePlatform', { ...existingSSHHostPlatforms, [hostname]: targetPlatform }, vscode.ConfigurationTarget.Global);
+            }
+        } catch (error) {
+            this.logService.error('Error updating remotePlatform configuration', error);
+        }
+    }
```

Comment:
maybe move it to remote.ts or extension.ts not really a critical path of hostservice, just some config update